jobbergate-api/jobbergate_api/apps/job_scripts/routers.py

```diff
@@ -287,21 +288,32 @@ async def job_script_update(job_script_id: int, job_script: JobScriptUpdateReque
     update_query = (
         job_scripts_table.update()
         .where(job_scripts_table.c.id == job_script_id)
-        .values(job_script.dict(exclude_unset=True))
+        .values(
+            job_script.dict(exclude_unset=True, exclude={"job_script_data_as_string"}),
+        )
         .returning(job_scripts_table)
     )
     try:
         result = await database.fetch_one(update_query)
     except INTEGRITY_CHECK_EXCEPTIONS as e:
         raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail=str(e))
+    else:
+        if job_script.job_script_data_as_string:
+            s3man_jobscripts.put(
+                job_script.job_script_data_as_string,
+                job_script_id,
+            )
 
     if result is None:
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND,
             detail=f"JobScript with id={job_script_id} not found.",
         )
 
-    return result
+    job_script_response = dict(result)
+    job_script_response["job_script_data_as_string"] = s3man_jobscripts.get_s3_object_as_string(job_script_id)
+
+    return job_script_response
```

Comment:
Theorically it a bit strange for me the route method is PUT. If we throw 404 then it should be PATCH, shouldn't?

Reply:
We discussed this separately. Maybe we address this in the future (conceptual difference between PUT and PATCH). Let's just move forward as-is for now.